lib/auth/user_gateway.dart

```diff
@@ -50,24 +56,36 @@ class UserGateway {
 
 class User {
   final String id;
-  final String displayName;
-  final String photoUrl;
   final String email;
   final bool emailVerified;
+  final String displayName;
+  final String photoUrl;
+  final String phoneNumber;
+  final bool disabled;
+  final DateTime tokensValidAfterTime;
 
   User.fromMap(Map<String, dynamic> map)
       : id = map['localId'],
         displayName = map['displayName'],
         photoUrl = map['photoUrl'],
         email = map['email'],
-        emailVerified = map['emailVerified'];
+        emailVerified = map['emailVerified'],
+        phoneNumber = map['phoneNumber'],
+        disabled = map['disabled'] ?? false,
+        tokensValidAfterTime = map['validSince'] == null
+            ? null
+            : DateTime.fromMillisecondsSinceEpoch(
+                int.parse(map['validSince']) * 1000);
 
   Map<String, dynamic> toMap() => {
         'localId': id,
         'displayName': displayName,
         'photoUrl': photoUrl,
         'email': email,
         'emailVerified': emailVerified,
+        'phoneNumber': phoneNumber,
+        'disabled': disabled,
```

Comment:
Is this coming from Firebase?

Reply:
Yes, the firebase sdk in dart includes phoneNumber. Not `disabled` though, that's just a nice feature to have.

Reply:
I was specifically asking about disabled. What does it do? Why is it nice?

Reply:
My assumption was that if you disable an account on firebase auth's console, disabled would be true

Reply:
So you added the field without checking if Firebase is even setting it?
It doesn't even make any sense, if the account is disabled you probably don't even get a token.

Reply:
I'll remove them. My understanding was if the user has an account on your app for example, but the account is banned/disabled, the token would be successful but the app would show a page with a warning that the account has been disabled, etc..

Reply:
I would ask that you test these things before submitting code blindly.
Please understand that I'm spending quite a bit of time I don't really have reviewing this code. At this point it would've been quicker to just implement it myself.